### PR TITLE
Non behavioral cleanup to the lexer prior to raw string work.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This shoudl be reviewed with whitespace changes off.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs [hunk 1]
```diff
@@ -17,86 +17,82 @@ namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
 {
     internal partial class Lexer
     {
-        private void ScanStringLiteral(ref TokenInfo info, bool allowEscapes = true)
+        private void ScanStringLiteral(ref TokenInfo info, bool inDirective)
         {
             var quoteCharacter = TextWindow.PeekChar();
-            if (quoteCharacter == '\'' || quoteCharacter == '"')
+            Debug.Assert(quoteCharacter == '\'' || quoteCharacter == '"');
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs:23]: no need for if check here.

File: src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs [hunk 2]
```diff
@@ -294,68 +290,69 @@ internal static SyntaxToken RescanInterpolatedString(InterpolatedStringExpressio
 
         private class InterpolatedStringScanner
         {
-            public readonly Lexer lexer;
-            public bool isVerbatim;
-            public bool allowNewlines;
+            private readonly Lexer _lexer;
+            private bool _isVerbatim;
+            private bool _allowNewlines;
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs:295]: made private and renamed.

File: src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs [hunk 3]
```diff
@@ -17,86 +17,82 @@ namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
 {
     internal partial class Lexer
     {
-        private void ScanStringLiteral(ref TokenInfo info, bool allowEscapes = true)
+        private void ScanStringLiteral(ref TokenInfo info, bool inDirective)
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs:20]: renamed, inverted and made non-optional.

**Review by 333fred — APPROVED** [review]
LGTM (commit 3). @dotnet/roslyn-compiler for a second review.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Pull request merged**